pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroUtils.java

```diff
@@ -305,46 +301,23 @@ public static boolean isSingleValueField(Field field) {
     }
   }
 
-  @SuppressWarnings("unchecked")
-  public static void extractField(FieldSpec fieldSpec, GenericRecord from, GenericRow to) {
-    String fieldName = fieldSpec.getName();
-
-    // Handle the Map type
-    if (fieldName.endsWith(MAP_KEY_COLUMN_SUFFIX)) {
-      String avroFieldName = fieldName.substring(0, fieldName.length() - MAP_KEY_COLUMN_SUFFIX.length());
-      Map map = (Map) from.get(avroFieldName);
-      if (map != null) {
-        // Sort the keys so that the order is deterministic
-        TreeSet sortedKeys = new TreeSet(map.keySet());
-        to.putField(fieldName, RecordReaderUtils.convert(fieldSpec, sortedKeys));
-        return;
-      }
-    } else if (fieldName.endsWith(MAP_VALUE_COLUMN_SUFFIX)) {
-      String avroFieldName = fieldName.substring(0, fieldName.length() - MAP_VALUE_COLUMN_SUFFIX.length());
-      Map map = (Map) from.get(avroFieldName);
-      if (map != null) {
-        // Sort the keys so that the order is deterministic
-        TreeMap sortedMap = new TreeMap<>(map);
-        to.putField(fieldName, RecordReaderUtils.convert(fieldSpec, sortedMap.values()));
-        return;
-      }
-    }
-    to.putField(fieldName, RecordReaderUtils.convert(fieldSpec, convert(fieldSpec, from.get(fieldName))));
-  }
-
   /**
-   * Converts the value based on the given field spec.
+   * Converts the value to a single-valued value or a multi-valued value
    */
-  public static Object convert(FieldSpec fieldSpec, @Nullable Object value) {
-    if (fieldSpec.isSingleValueField()) {
-      return handleSingleValue(value);
+  public static Object convert(Object value) {
+    Object convertedValue;
+    if (value instanceof Collection) {
```

Comment:
Can value be an instance of primitive array?

Reply:
In avro, it cannot be